src/ray/core_worker/core_worker.cc

```diff
@@ -238,6 +248,24 @@ void CoreWorkerProcess::RunTaskExecutionLoop() {
   instance_.reset();
 }
 
+void CoreWorkerProcess::RunStatsService() {
+  absl::MutexLock lock(&CoreWorkerProcess::stats_mutex_);
+  if (CoreWorkerProcess::stats_thread_) {
+    return;
+  }
+  // Initialize stats.
+  const ray::stats::TagsType global_tags = {{ray::stats::ComponentKey, "core_worker"},
+                                            {ray::stats::VersionKey, "0.9.0.dev0"}};
+
+  ray::stats::Init(global_tags, RayConfig::instance().metrics_agent_port(),
+                   CoreWorkerProcess::stats_io_service_);
+  CoreWorkerProcess::stats_thread_.reset(
+      new std::thread([]() { CoreWorkerProcess::stats_io_service_.run(); }));
+  // TODO(lingxuan.zlx): Should shutdown if opencensus disabled.
```

Comment:
Does this mean we can enable & disable stats on the fly? Why do we need this feature?

Reply:
> Does this mean we can enable & disable stats on the fly? Why do we need this feature?

An explaintion given in the other comment. In short, we create stats thread exactly once.